scene/resources/syntax_highlighter.h

```diff
@@ -142,4 +143,31 @@ class CodeHighlighter : public SyntaxHighlighter {
 	Color get_member_variable_color() const;
 };
 
+///////////////////////////////////////////////////////////////////////////////
+
+class GDShaderSyntaxHighlighter : public CodeHighlighter {
+	GDCLASS(GDShaderSyntaxHighlighter, CodeHighlighter)
+
+private:
+	Vector<Point2i> disabled_branch_regions;
+	Color disabled_branch_color;
+
+protected:
+	static void _bind_methods();
+
+public:
+	virtual Dictionary _get_line_syntax_highlighting_impl(int p_line) override;
+
+	void add_disabled_branch_region(int p_from_line, int p_to_line);
+	void remove_disabled_branch_region(int p_from_line);
+	bool has_disabled_branch_region(int p_from_line) const;
+
+	void set_disabled_branch_regions(const Vector<String> &p_regions);
```

Comment:
Ditto here, or as it editor only can remove these methods if we have no plans to use them.